docs/terraform-code-structure.md

```diff
@@ -0,0 +1,112 @@
+# Terraform modules structure
+
+## Filenames <!-- omit in toc -->
+
+* [`0-init.tf`](#0-inittf)
```

Comment:
Why you have this numbers in naming?  0,1,8,9?

Reply:
For sort logically. 

Imagine that you have many files in your module (simple case [exist now in terraform/aws/addons](https://github.com/shalb/cluster.dev/tree/df78211dff8de8f3ab31c7bea61f03b5827734a4/terraform/aws/addons))


```
View now       | View in IDE     | View in Github  |
-------------- | --------------- | --------------- |
argocd.yaml    | 0-init.tf       | README.md       |
init.tf        | 1-main.tf       | 0-init.tf       |
file.sh        | 2-too-logic.tf  | 1-main.tf       |
main.tf        | 3-more-logic.tf | 2-too-logic.tf  |
more-logic.tf  | 8-outputs.tf    | 3-more-logic.tf |
README.md      | 9-vars.tf       | 8-outputs.tf    |
outputs.tf     | argocd.yaml     | 9-vars.tf       |
too-logic.tf   | file.sh         | argocd.yaml     |
something.yaml | README.md       | file.sh         |
vars.tf        | something.yaml  | something.yaml  |
```

And we can compose files with logic (`main.tf`, `too-logic.tf` `more-logic.tf`) as it will be really executed/depends. Or as we'd like

For ex:
`1-main.tf`, `2-too-logic.tf`, `3-more-logic.tf`
`1-more-logic.tf`, `2-too-logic.tf`, `3-main.tf` 
`1-too-logic.tf`, `2-more-logic.tf`, `3-main.tf`

Reply:
outputs.tf and variables.tf - is the standart ways across all modules in internet :).
I'd like to keep thisn naming as is.
So lets approve this without numbers in front.
Also for templates I thinks we should use dedicated /templates folder